pkg/sources/github/github.go

```diff
@@ -833,6 +826,41 @@ func (s *Source) scan(ctx context.Context, installationClient *github.Client, ch
 	return nil
 }
 
+func (s *Source) cloneAndScanRepo(ctx context.Context, client *github.Client, repoURL string, chunksChan chan *sources.Chunk) (time.Duration, error) {
+	var duration time.Duration
+
+	ctx.Logger().V(2).Info("attempting to clone repo")
+	path, repo, err := s.cloneRepo(ctx, repoURL, client)
+	defer func(path string) {
+		if path != "" {
+			_ = os.RemoveAll(path)
+		}
+	}(path)
```

Comment:
Thank you for aggressively preserving the order of operations here, but I think you can just move `defer os.RemoveAll(path)` to after the error check :) (@mcastorina that's what you did elsewhere for something, right?)

Reply:
ah yes, a similar refactor for gitlab in #2345